src/hooks/a11y/useTrapFocus.js

```diff
@@ -0,0 +1,80 @@
+import { useState, useEffect } from 'react'
+
+const FOCUSABLE_ELEMENTS_STRING =
+  'a[href], area[href], input:not([disabled]), select:not([disabled]), textarea:not([disabled]), button:not([disabled]), iframe, object, embed, [tabindex="0"], [contenteditable]'
+
+/**
+ * Helper function returns a list of focusable children of the passed element
+ *
+ * @param {HTMLElement} el - the container element
+ */
+function getFocusableElements(el) {
+  if (!el) {
+    return []
+  }
+
+  const focusableElements = el.querySelectorAll(FOCUSABLE_ELEMENTS_STRING)
+
+  return Array.prototype.slice.call(focusableElements)
+}
+
+/**
+ * Hook will trap focus within a component
+ *
+ * @param {React.MutableRefObject<HTMLElement | null>} ref - the component's ref
+ * @param {boolean} isActive - if true, the focus trap will be activated
+ *
+ * @return {void}
+ */
+function useTrapFocus(ref, isActive) {
+  const [focusedElemIdx, setFocusedElemIdx] = useState(0)
+
+  /**
+   * Handler for tab/shift+tab key presses and increments or
+   * decrements the `focusedElemIdx`.
+   *
+   * @param {number} lastIndex - the lastIndex that can be selected in the `elements`
+   * array
+   *
+   * @return {(e: KeyboardEvent) => void}
+   */
+  const handleTrapTab = (lastIndex) => {
```

Comment:
Non-blocker but opening then clicking to establish focus in the modal and making the first key press a shift-tab focuses stuff in the background

Reply:
ah, good catch!, I'll take a look